src/Adapter/Image/ImageRetriever.php

```diff
@@ -165,47 +170,56 @@ public function getImage($object, $id_image)
         $urls = [];
         $image_types = ImageType::getImagesTypes($type, true);
         $generateHighDpiImages = (bool) Configuration::get('PS_HIGHT_DPI');
-        $ext = 'jpg';
 
-        // Get path of original uploaded image we will use to get thumbnails
+        // Get path of original uploaded image we will use to get thumbnails (original image is always .jpg)
         $originalImagePath = implode(DIRECTORY_SEPARATOR, [
             $imageFolderPath,
-            $id_image . '.' . $ext,
+            $id_image . '.jpg',
         ]);
 
+        $rewriteLink = !empty($object->link_rewrite) ? $object->link_rewrite : $object->name;
         foreach ($image_types as $image_type) {
-            // Final path thumbnail in our size
-            $thumbnailPath = implode(DIRECTORY_SEPARATOR, [
-                $imageFolderPath,
-                $id_image . '-' . $image_type['name'] . '.' . $ext,
-            ]);
+            $additionalSources = [];
+
+            if (!$this->isMultipleImageFormatFeatureActive || Configuration::get('PS_ADDITIONAL_IMAGE_JPG')) {
```

Comment:
can you use dependendy injection for the `Configuration` service ?

Reply:
The problem I have is that ImageRetriever is instanciated manually everywhere it's used, and adding a construct parameter  in those circumstances is a BC break

Reply:
😢